.generator/src/generator/templates/model_generic.j2

```diff
@@ -54,16 +54,22 @@ class {{ name }}(ModelNormal):
     }
 {%- endif %}
 
-{%- if model.get("additionalProperties", false) is not false %}
+{%- set additionalProperties = model.get("additionalProperties") %}
+{#- Only add additional_properties_type property if additionalProperties is disabled or has a custom type #}
+{%- if (additionalProperties and additionalProperties is not true) or additionalProperties is false %}
```

Comment:
```suggestion
{%- if (additionalProperties is false) or (additionalProperties and additionalProperties is not true) %}
```
Would be nice if the order of the expression matched the order of the comment (not required though).